modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -435,14 +443,25 @@ const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>): void => {
   }
 
   // Never split the body or blocks that we can't split like noneditable host elements
-  if (parentBlock === editor.getBody() || !canSplitBlock(dom, parentBlock)) {
+  if (!inRootAndLastOrCef && (parentBlock === editor.getBody() || !canSplitBlock(dom, parentBlock))) {
     return;
   }
   const parentBlockParent = parentBlock.parentNode;
 
   // Insert new block before/after the parent block depending on caret location
   let newBlock: Element;
-  if (CaretContainer.isCaretContainerBlock(parentBlock)) {
+  if (inRootAndLastOrCef) {
+    newBlock = createNewBlock(newBlockName);
+    child.fold(
+      () => {
+        Insert.append(start, SugarElement.fromDom(newBlock));
+      },
+      (child) => {
+        Insert.before(child, SugarElement.fromDom(newBlock));
+      }
+    );
+    editor.selection.setCursorLocation(newBlock, 0);
```

Comment:
What's this code doing that the code further down this `if` block doesn't already? Can we just adjust the order of the conditions somehow to achieve the same effect?
https://github.com/tinymce/tinymce/blob/873133a8529034434a414225dd2d88522d3bad8d/modules/tinymce/src/core/main/ts/newline/InsertBlock.ts#L452-L457

Reply:
It uses the newBlockName instead of going on default, which is important as otherwise it attempts to clone the next element when one exist, which is a CEF, giving the new element that same property.